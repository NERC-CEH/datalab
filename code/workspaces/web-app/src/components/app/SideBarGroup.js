import React from 'react';
import Typography from '@mui/material/Typography';
import withStyles from '@mui/styles/withStyles';

const styles = theme => ({
  sidebarGroup: {
    padding: `${theme.spacing(3)} 0`,
    '& + &': {
      borderTop: `1px solid ${theme.palette.divider}`,
    },
  },
  title: {
    marginLeft: theme.spacing(2),
  },
});

function SideBarGroup({ classes, children, title }) {
  // Handles case where might not have permission for anything in group.
  if (!children) return null;

  return (
    <div className={classes.sidebarGroup}>
      {title ? <Typography className={classes.title} variant='h6'>{title}</Typography> : null}
      {children}
    </div>
  );
}

export default withStyles(styles)(SideBarGroup);
