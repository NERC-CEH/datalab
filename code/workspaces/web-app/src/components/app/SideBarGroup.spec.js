import React from 'react';
import { render } from '@testing-library/react';
import SideBarGroup from './SideBarGroup';

describe('SideBarGroup', () => {
  describe('when no children provided', () => {
    it('renders nothing when no title provided', () => {
      const wrapper = render(<SideBarGroup />);
      expect(wrapper.container).toMatchSnapshot();
    });

    it('renders nothing when title provided', () => {
      const wrapper = render(<SideBarGroup title='Group Title' />);
      expect(wrapper.container).toMatchSnapshot();
    });
  });

  describe('when children provided', () => {
    it('renders children when no title provided', () => {
      const wrapper = render(<SideBarGroup><div>test div content</div></SideBarGroup>);
      expect(wrapper.container).toMatchSnapshot();
    });

    it('renders children and title when title provided', () => {
      const wrapper = render(<SideBarGroup title='Test Group'><div>test div content</div></SideBarGroup>);
      expect(wrapper.container).toMatchSnapshot();
    });
  });
});
