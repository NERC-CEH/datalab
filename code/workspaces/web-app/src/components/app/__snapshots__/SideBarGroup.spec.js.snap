// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarGroup when children provided renders children and title when title provided 1`] = `
<div
  className="SideBarGroup-sidebarGroup-1"
>
  <WithStyles(ForwardRef(Typography))
    className="SideBarGroup-title-2"
    variant="h6"
  >
    Test Group
  </WithStyles(ForwardRef(Typography))>
  <div>
    test div content
  </div>
</div>
`;

exports[`SideBarGroup when children provided renders children when no title provided 1`] = `
<div
  className="SideBarGroup-sidebarGroup-1"
>
  <div>
    test div content
  </div>
</div>
`;

exports[`SideBarGroup when no children provided renders nothing when no title provided 1`] = `""`;

exports[`SideBarGroup when no children provided renders nothing when title provided 1`] = `""`;
